includes/util.py

```
@@ -0,0 +1,135 @@
+        while True:
+            data = f.read(BUF_SIZE)
+            if not data:
+                break
+            sha1.update(data)
+    return sha1.hexdigest()
+
+def generate_testfile(sourcefile,destfile,minsize):
+    ''' Make tempfiles that are concatenated repeatedly until the file is big enough '''
+    srcsize = os.path.getsize(sourcefile)
+    if srcsize == 0:
+        print(f"Error: Sourcefile '{filename}' is empty, cannot generate testfiles.")
```

Ruff (0.14.13)

Undefined name `filename`

(F821)